### PR TITLE
xfsprogs: 4.19 -> 5.10

### PR DESCRIPTION
**Pull request by ajs124**
###### Motivation for this change
xfsprogs was outdated by several years and does not appear to have an (active) maintainer.

Closes #107010

cc @TredwellGit @Mic92

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lovesegfault** [comment]
Seems like this should target staging?

File: pkgs/development/libraries/libinih/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  version = "52";
+
+  src = fetchFromGitHub {
+    owner = "benhoyt";
+    repo = "inih";
+    rev = "r${version}";
+    sha256 = "0lsvm34zabvi1xlximybzvgc58zb90mm3b9babwxlqs05jy871m4";
+  };
+
+  mesonFlags = [
```
Comment (TredwellGit) [inline pkgs/development/libraries/libinih/default.nix:14]: Why did you remove `"-Dwith_INIReader=true"`?

File: pkgs/development/libraries/libinih/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "-Ddistro_install=true"
+  ];
+
+  nativeBuildInputs = [
+    meson
+    ninja
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = "";
```
Comment (TredwellGit) [inline pkgs/development/libraries/libinih/default.nix:25]: Why did you remove the homepage and changelog?

**Comment by ajs124** [comment]
ofborg says <500 rebuilds, so I guess it should be fine?

File: pkgs/development/libraries/libinih/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    description = "A simple .INI file parser written in C";
+    license = licenses.bsd3;
+    platforms = platforms.linux;
```
Comment (TredwellGit) [inline pkgs/development/libraries/libinih/default.nix:28]: Why `platforms.linux` instead of `platforms.all`?

**Review by TredwellGit — CHANGES_REQUESTED** [review]
I think https://github.com/NixOS/nixpkgs/pull/107010 is better for inih.

File: pkgs/development/libraries/libinih/default.nix (hunk 1 above)
Comment (ajs124) [reply pkgs/development/libraries/libinih/default.nix:14]: It only occured to me to search the tracker for xfsprogs after I already did all these things, but I can incorporate your changes.

File: pkgs/development/libraries/libinih/default.nix (hunk 2 above)
Comment (ajs124) [reply pkgs/development/libraries/libinih/default.nix:25]: As explained above, I didn't remove them, I just did not add them.

File: pkgs/development/libraries/libinih/default.nix (hunk 3 above)
Comment (ajs124) [reply pkgs/development/libraries/libinih/default.nix:28]: Because I can't test anything else and I therefore cannot support them as a maintainer.

**Comment by ajs124** [comment]
@TredwellGit we can also merge #107010 and I can rebase on that, but it should have a maintainer.

**Comment by TredwellGit** [comment]
Let me add myself as a maintainer.

**Comment by lovesegfault** [comment]
Result of `nixpkgs-review pr 107597` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages marked as broken and skipped:</summary>
  <ul>
    <li>bareos</li>
    <li>gnome3.gnome-books</li>
    <li>gnome3.gnome-documents</li>
    <li>xmonad_log_applet</li>
    <li>xmonad_log_applet_gnome3</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>samba4Full</li>
    <li>scylladb</li>
  </ul>
</details>
<details>
  <summary>122 packages built:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>almanah</li>
    <li>atom</li>
    <li>atom-beta</li>
    <li>bubblemail</li>
    <li>calls</li>
    <li>cantata</li>
    <li>ceph</li>
    <li>ceph-client</li>
    <li>charliecloud</li>
    <li>chrome-gnome-shell</li>
    <li>cinnamon.cinnamon-common</li>
    <li>cinnamon.cinnamon-control-center</li>
    <li>cinnamon.cinnamon-gsettings-overrides</li>
    <li>cinnamon.cinnamon-screensaver</li>
    <li>clementine</li>
    <li>clementineUnfree</li>
    <li>disnix</li>
    <li>disnixos</li>
    <li>docker</li>
    <li>docker-gc</li>
    <li>docker_18_09</li>
    <li>dropbox-cli</li>
    <li>dydisnix</li>
    <li>dysnomia</li>
    <li>elementary-planner</li>
    <li>empathy</li>
    <li>enlightenment.enlightenment</li>
    <li>evolution-data-server (evolution_data_server)</li>
    <li>evolution-ews</li>
    <li>far2l</li>
    <li>feedreader</li>
    <li>folks</li>
    <li>gfbgraph</li>
    <li>glusterfs</li>
    <li>gnome-multi-writer</li>
    <li>gnome-online-accounts</li>
    <li>gnome-photos</li>
    <li>gnome-recipes</li>
    <li>gnome2.gvfs</li>
    <li>gnome3.bijiben</li>
    <li>gnome3.cheese</li>
    <li>gnome3.evolution</li>
    <li>gnome3.file-roller</li>
    <li>gnome3.geary</li>
    <li>gnome3.gnome-applets</li>
    <li>gnome3.gnome-boxes</li>
    <li>gnome3.gnome-calendar</li>
    <li>gnome3.gnome-contacts</li>
    <li>gnome3.gnome-control-center</li>
    <li>gnome3.gnome-disk-utility</li>
    <li>gnome3.gnome-flashback</li>
    <li>gnome3.gnome-initial-setup</li>
    <li>gnome3.gnome-maps</li>
    <li>gnome3.gnome-music</li>
    <li>gnome3.gnome-online-miners</li>
    <li>gnome3.gnome-panel</li>
    <li>gnome3.gnome-session</li>
    <li>gnome3.gnome-shell</li>
    <li>gnome3.gnome-software</li>
    <li>gnome3.gnome-terminal</li>
    <li>gnome3.gnome-todo</li>
    <li>gnome3.gnome-tweak-tool</li>
    <li>gnome3.gnome-user-share</li>
    <li>gnome3.grilo-plugins</li>
    <li>gnome3.gvfs</li>
    <li>gnome3.libgdata</li>
    <li>gnome3.libzapojit</li>
    <li>gnome3.nautilus</li>
    <li>gnome3.nautilus-python</li>
    <li>gnome3.pomodoro</li>
    <li>gnome3.shotwell</li>
    <li>gnome3.totem</li>
    <li>gnome3.tracker-miners</li>
    <li>gnomeExtensions.easyScreenCast</li>
    <li>gnomeExtensions.gsconnect</li>
    <li>gnomeExtensions.night-theme-switcher</li>
    <li>hal-flash</li>
    <li>libceph</li>
    <li>libguestfs</li>
    <li>libguestfs-with-appliance</li>
    <li>libinih</li>
    <li>libvmi</li>
    <li>libxfs</li>
    <li>mailnagWithPlugins</li>
    <li>mate.mate-utils</li>
    <li>nfs-ganesha</li>
    <li>out-of-tree</li>
    <li>pantheon.elementary-calendar</li>
    <li>pantheon.elementary-greeter</li>
    <li>pantheon.elementary-session-settings</li>
    <li>pantheon.extra-elementary-contracts</li>
    <li>pantheon.wingpanel-indicator-datetime</li>
    <li>pantheon.wingpanel-with-indicators</li>
    <li>pipework</li>
    <li>psensor</li>
    <li>python37Packages.guestfs</li>
    <li>python37Packages.jupyter-repo2docker</li>
    <li>python38Packages.guestfs</li>
    <li>python38Packages.jupyter-repo2docker</li>
    <li>python39Packages.guestfs</li>
    <li>python39Packages.jupyter-repo2docker</li>
    <li>qemu_xen (qemu_xen_4_10)</li>
    <li>qubes-core-vchan-xen</li>
    <li>rabbitvcs</li>
    <li>rapid-photo-downloader</li>
    <li>spaceFM</li>
    <li>udiskie</li>
    <li>udisks</li>
    <li>usermount</li>
    <li>vagrant</li>
    <li>vifm-full</li>
    <li>xen</li>
    <li>xen-slim</li>
    <li>xfce.gigolo</li>
    <li>xfce.gvfs</li>
    <li>xfce.thunar (xfce.thunar-bare)</li>
    <li>xfce.thunar-archive-plugin</li>
    <li>xfce.thunar-dropbox-plugin</li>
    <li>xfce.xfdesktop</li>
    <li>xfsprogs</li>
    <li>xfstests</li>
  </ul>
</details>

**Comment by TredwellGit** [comment]
Okay, please merge https://github.com/NixOS/nixpkgs/pull/107010 then rebase.

**Comment by ajs124** [comment]
@TredwellGit  done

@lovesegfault I'll see if I can figure out what's happening with samba4Full (and scylladb). Are they actually working on master?

**Comment by lovesegfault** [comment]
> @TredwellGit done
> 
> @lovesegfault I'll see if I can figure out what's happening with samba4Full (and scylladb). Are they actually working on master?

Let me check

**Comment by lovesegfault** [comment]
> > @TredwellGit done
> > @lovesegfault I'll see if I can figure out what's happening with samba4Full (and scylladb). Are they actually working on master?
> 
> Let me check

Both are broken in master.

**Comment by ajs124** [comment]
> > > @TredwellGit done
> > > @lovesegfault I'll see if I can figure out what's happening with samba4Full (and scylladb). Are they actually working on master?
> > 
> > 
> > Let me check
> 
> Both are broken in master.

That's good. I mean, it isn't, but that means that this PR is not at fault.

I'll leave this open, in case anyone wants to review it or has any objections, e.g. if we should ship a newer xfsprogs than our current default kernel release or something like that.

In case no-one requests any changes, I'll merge this in 7 days (in my timezone), so on 2021-01-02. Feel free to merge it before that, but seeing as this hasn't been updated since December of 2018, there's probably no need to rush.

File: pkgs/tools/filesystems/xfsprogs/default.nix [hunk 4]
```diff
@@ -26,26 +18,23 @@ stdenv.mkDerivation rec {
...
-    (gentooPatch "xfsprogs-4.15.0-sharedlibs.patch" "0bv2naxpiw7vcsg8p1v2i47wgfda91z1xy1kfwydbp4wmb4nbyyv")
-    (gentooPatch "xfsprogs-4.15.0-docdir.patch" "1srgdidvq2ka0rmfdwpqp92fapgh53w1h7rajm4nnby5vp2v8dfr")
-    (gentooPatch "xfsprogs-4.9.0-underlinking.patch" "1r7l8jphspy14i43zbfnjrnyrdm4cpgyfchblascxylmans0gci7")
-  ];
-
   preConfigure = ''
-    sed -i Makefile -e '/cp include.install-sh/d'
+    for file in scrub/{xfs_scrub_all.cron.in,xfs_scrub@.service.in,xfs_scrub_all.service.in}; do
+      substituteInPlace "$file" \
+        --replace '@sbindir@' '/run/current-system/sw/bin'
```
Comment (Mic92) [inline pkgs/tools/filesystems/xfsprogs/default.nix:29]: Maybe add a comment why you use an `/run/current-system` rather than a path to nix store.

Comment (ajs124) [reply pkgs/tools/filesystems/xfsprogs/default.nix:30]: How about this?

Comment (ajs124) [reply pkgs/tools/filesystems/xfsprogs/default.nix:30]: @Mic92

**Comment by ajs124** [comment]
After asking upstream, I think we should add a release notes entry to this, which I will write and push shortly.

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107597` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages marked as broken and skipped:
</summary>
<ul>
<li>gnome3.gnome-books</li>
<li>gnome3.gnome-documents</li>
<li>xmonad_log_applet</li>
</ul>
</details>
<details>
<summary>
2 packages blacklisted:
</summary>
<ul>
<li>tests.nixos-functions.nixos-test</li>
<li>tests.nixos-functions.nixosTest-test</li>
</ul>
</details>
<details>
<summary>
117 packages built:
</summary>
<ul>
<li>adapta-gtk-theme</li>
<li>almanah</li>
<li>atom</li>
<li>atom-beta</li>
<li>bubblemail</li>
<li>calls</li>
<li>cantata</li>
<li>charliecloud</li>
<li>chrome-gnome-shell</li>
<li>cinnamon.cinnamon-common</li>
<li>cinnamon.cinnamon-control-center</li>
<li>cinnamon.cinnamon-gsettings-overrides</li>
<li>cinnamon.cinnamon-screensaver</li>
<li>clementine</li>
<li>clementineUnfree</li>
<li>disnix</li>
<li>disnixos</li>
<li>docker (docker-edge ,docker_19_03)</li>
<li>docker-gc</li>
<li>docker_18_09</li>
<li>dropbox-cli</li>
<li>dydisnix</li>
<li>dysnomia</li>
<li>elementary-planner</li>
<li>empathy (gnome3.empathy)</li>
<li>enlightenment.enlightenment</li>
<li>evolution-data-server (gnome3.evolution-data-server ,gnome3.evolution_data_server)</li>
<li>evolution-ews</li>
<li>far2l</li>
<li>feedreader</li>
<li>folks (gnome3.folks)</li>
<li>gfbgraph (gnome3.gfbgraph)</li>
<li>glusterfs</li>
<li>gnome-multi-writer</li>
<li>gnome-online-accounts (gnome3.gnome-online-accounts ,gnome3.gnome_online_accounts)</li>
<li>gnome-photos (gnome3.gnome-photos)</li>
<li>gnome-recipes</li>
<li>gvfs (gnome2.gvfs)</li>
<li>gnome3.bijiben (gnome3.gnome-notes)</li>
<li>gnome3.cheese</li>
<li>gnome3.evolution</li>
<li>gnome3.file-roller</li>
<li>gnome3.geary</li>
<li>gnome3.gnome-applets</li>
<li>gnome3.gnome-boxes</li>
<li>gnome3.gnome-calendar</li>
<li>gnome3.gnome-contacts</li>
<li>gnome3.gnome-control-center (gnome3.gnome_control_center)</li>
<li>gnome3.gnome-disk-utility</li>
<li>gnome3.gnome-flashback</li>
<li>gnome3.gnome-initial-setup</li>
<li>gnome3.gnome-maps</li>
<li>gnome3.gnome-music</li>
<li>gnome3.gnome-online-miners</li>
<li>gnome3.gnome-panel</li>
<li>gnome3.gnome-session (gnome3.gnome_session)</li>
<li>gnome3.gnome-shell (gnome3.gnome_shell)</li>
<li>gnome3.gnome-software</li>
<li>gnome3.gnome-terminal (gnome3.gnome_terminal)</li>
<li>gnome3.gnome-todo</li>
<li>gnome3.gnome-tweaks (gnome3.gnome-tweak-tool)</li>
<li>gnome3.gnome-user-share</li>
<li>grilo-plugins (gnome3.grilo-plugins)</li>
<li>gnome3.gvfs</li>
<li>libgdata (gnome3.libgdata)</li>
<li>libzapojit (gnome3.libzapojit)</li>
<li>gnome3.nautilus</li>
<li>gnome3.nautilus-python</li>
<li>gnome3.pomodoro</li>
<li>shotwell (gnome3.shotwell)</li>
<li>gnome3.totem</li>
<li>tracker-miners (gnome3.tracker-miners)</li>
<li>gnomeExtensions.easyScreenCast</li>
<li>gnomeExtensions.gsconnect</li>
<li>gnomeExtensions.night-theme-switcher</li>
<li>hal-flash</li>
<li>libguestfs</li>
<li>libvmi</li>
<li>libxfs</li>
<li>mailnagWithPlugins</li>
<li>mate.mate-utils</li>
<li>nfs-ganesha</li>
<li>out-of-tree</li>
<li>pantheon.elementary-calendar</li>
<li>pantheon.elementary-greeter</li>
<li>pantheon.elementary-session-settings</li>
<li>pantheon.extra-elementary-contracts</li>
<li>pantheon.wingpanel-indicator-datetime</li>
<li>pantheon.wingpanel-with-indicators</li>
<li>pipework</li>
<li>psensor</li>
<li>python37Packages.guestfs</li>
<li>python37Packages.jupyter-repo2docker</li>
<li>python38Packages.guestfs</li>
<li>python38Packages.jupyter-repo2docker</li>
<li>python39Packages.guestfs</li>
<li>python39Packages.jupyter-repo2docker</li>
<li>qemu_xen (qemu_xen_4_10)</li>
<li>qubes-core-vchan-xen</li>
<li>rabbitvcs</li>
<li>rapid-photo-downloader</li>
<li>spaceFM</li>
<li>udiskie</li>
<li>udisks (udisks2)</li>
<li>usermount</li>
<li>vagrant</li>
<li>vifm-full</li>
<li>xen (xenPackages.xen-vanilla ,xenPackages.xen_4_10-vanilla ,xen_4_10)</li>
<li>xen-slim (xenPackages.xen-slim ,xenPackages.xen_4_10-slim ,xen_4_10-slim)</li>
<li>xfce.gigolo</li>
<li>xfce.gvfs</li>
<li>xfce.thunar (xfce.thunar-bare ,xfce.thunar-build ,xfce.thunarx-2-dev)</li>
<li>xfce.thunar-archive-plugin</li>
<li>xfce.thunar-dropbox-plugin</li>
<li>xfce.xfdesktop</li>
<li>xfsprogs</li>
<li>xfstests</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

bubblemail:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>
mailnagWithPlugins:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python37Packages.guestfs:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.guestfs:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.guestfs:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>
